architecture/additional_concepts/dynamic_admission_controllers.adoc

```diff
@@ -0,0 +1,473 @@
+external admission webhook to do this validation before the pod is instantiated 
+and prevent the pod from being created.
+
+Some common use-cases include:
+
+* Mutating resources, for example such as to inject side-car containers into pods. 
+* Restricting projects, for example, to block some resources from a project. 
+* Complex CustomResource validation, such as performing complex validation on dependent fields.
+* Security response, for example, preventing certain SHAs from running.
+
+External admission webhooks allow a cluster administrator to add mutating and validating admission plugins 
+to the admission chain of the API server recompiling.
+
+[IMPORTANT]
+====
+*External Admission Webhooks* are Technology Preview featuresonly.
+ifdef::openshift-enterprise[]
+Technology Preview features are not supported with Red Hat production service
+level agreements (SLAs), might not be functionally complete, and Red Hat does
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+The external admission webhooks support TLS because the admission control operations require high-security.
+
+Whenever a request comes in, the Generic Admission Webhook admission controller 
+gets the list of external admission webhooks from
+xref:architecture-additional-concepts-dynamic-admission-webhooks-writing[`externalAdmissionHookConfiguration`]
+objects and call them in parallel: 
+
+* If *all* of the external admission webhooks approve the admission
+request, the admission chain continues. 
+
+* If *any* of the external admission
+webhooks deny the admission request, the admission request will be denied, and
+the reason for doing so will be based on the _first_ external admission webhook
+denial reason. 
++
+This means if there is more than one external admission webhook
+that denied the admission request, only the first will be returned to the
+user.
+
+* If there is an error encountered when calling an external admission
+webhook, that request is ignored and will not be used to approve/deny the
+admission request.
+
+[NOTE]
+====
+The admission chain depends solely on the order of the
+`--admission-control` option passed to `kube-apiserver`.
+====
+
+The communication between the webhook admission controller and the Generic Admission Webhook admission controller, needs to be
+secured using  TLS. You need to generate a CA certoificate and use the certificate to sign the server certificate
+used by your webhook admission controller. The PEM-formatted CA certificate is supplied
+to the Generic Admission Webhook admission controller through the dynamic registration API
+`externaladmissionhookconfigurations.clientConfig.caBundle`.
```

Comment:
the whole type is called differently now. Compare the types.go link above and my shared documentation. Actually, there are two types: for validating and mutating webhooks.

Reply:
The preferred way is to use automatic cert generation https://docs.openshift.com/online/dev_guide/secrets.html#service-serving-certificate-secrets.